src/librustc/hir/map/collector.rs

```diff
@@ -118,7 +119,7 @@ impl<'a, 'hir> NodeCollector<'a, 'hir> {
     }
 
     pub(super) fn finalize_and_compute_crate_hash(self,
-                                                  crate_disambiguator: &str)
+                                                  crate_disambiguator: &Fingerprint)
```

Comment:
Usually we pass around `Fingerprint` by value. It implements `Copy` so that should be easy.